### PR TITLE
Increase TIMEOUT_FACTOR when running TSAN/MSAN/ASAN

### PR DESCRIPTION
**Pull request by fredyw**
Use the same `TIMEOUT_FACTOR` when running TSAN, MSAN, and ASAN since running the tests under sanitizers can take longer. This also renames `TSAN_TIMEOUT_FACTOR` to `TIMEOUT_FACTOR` to reflect the name.

Risk Level: low (tests only)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Review by alyssawilk — COMMENTED** [review]
LG modulo nits I'd like to fix while we're in here

File: test/integration/http_integration.h [hunk 1]
```diff
@@ -307,7 +307,7 @@ class HttpIntegrationTest : public BaseIntegrationTest {
   // Test sending and receiving large request and response bodies with autonomous upstream.
   void testGiantRequestAndResponse(
       uint64_t request_size, uint64_t response_size, bool set_content_length_header,
-      std::chrono::milliseconds timeout = 2 * TestUtility::DefaultTimeout * TSAN_TIMEOUT_FACTOR);
+      std::chrono::milliseconds timeout = 2 * TestUtility::DefaultTimeout * TIMEOUT_FACTOR);
```
[the remark this answers is not in the document]
Comment (fredyw) [reply test/integration/http_integration.h]: Done.

File: test/integration/idle_timeout_integration_test.cc [hunk 2]
```diff
@@ -118,7 +118,7 @@ class IdleTimeoutIntegrationTest : public HttpProtocolIntegrationTest {
 
   // TODO(htuch): This might require scaling for TSAN/ASAN/Valgrind/etc. Bump if
   // this is the cause of flakes.
-  static constexpr uint64_t IdleTimeoutMs = 300 * TSAN_TIMEOUT_FACTOR;
+  static constexpr uint64_t IdleTimeoutMs = 300 * TIMEOUT_FACTOR;
```
[the remark this answers is not in the document]
Comment (fredyw) [reply test/integration/idle_timeout_integration_test.cc:119]: Done.

**Review by alyssawilk — APPROVED** [review]
Thanks for the fix!

**Pull request merged**